web/src/views/DandisetLandingView/DandisetMain.vue

```diff
@@ -323,6 +382,11 @@ export default defineComponent({
       return moment(date).format('LL');
     }
 
+    function copy(value:string) {
+      const version = meta.value?.version === 'draft' ? meta.value?.identifier as string : meta.value?.id as string;
+      navigator.clipboard.writeText(value === 'dandiID' ? version : `https://doi.org/:${meta.value?.doi}`);
+    }
```

Comment:
In this situation, if `meta.value` is 	`undefined`, `undefined` or `https://doi.org/:undefined` will be copied to the clipboard. Since this is essentially conveying that the metadata needs to be present, why don't we just explicitly enforce that:

```suggestion
    function copy(value: string) {
      if (!meta.value) {
        throw new Error('metadata is undefined!');
      }

      const version = meta.value.version === 'draft' ? meta.value.identifier : meta.value.id;
      navigator.clipboard.writeText(value === 'dandiID' ? version : `https://doi.org/:${meta.value.doi}`);
    }
```

Reply:
added here: 92c257ba17bc8d31626e86a4b7f28767c51975ea